sdk/python/kfp/client/client.py

```diff
@@ -303,14 +303,10 @@ def _load_config(
                 namespace)
         return config
 
-    def _is_inverse_proxy_host(self, host):
+    def _is_inverse_proxy_host(self, host: str) -> bool:
         if host:
-            return re.match(r'\S+.googleusercontent.com/{0,1}$', host)
-        if re.match(r'\w+', host):
-            warnings.warn(
-                f'The received host is {host}, please include the full endpoint'
-                ' address (with ".(pipelines/notebooks).googleusercontent.com")'
-            )
```

Comment:
AFACT, this code is never reached. The only time that we reach line 309 is when `host` is `""` or `None`. Since both are falsey, this condition never evaluates as `True`. As a result, the subsequent warning on lines 310-313 is never executed.

We may want to handle `host` being `""` or `None` with a simple `else` aligned to the `if host` conditional, but the warning should probably be reworded in that case.

Reply:
Casting `re.Match` or `None` to a boolean and `-> bool` SGTM.